service/lib/agama/storage/config_conversions/to_json_conversions/encryption.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2024] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "agama/storage/config_conversions/to_json_conversions/base"
+require "agama/storage/config_conversions/to_json_conversions/luks1"
+require "agama/storage/config_conversions/to_json_conversions/luks2"
+require "agama/storage/config_conversions/to_json_conversions/pervasive_luks2"
+require "agama/storage/configs/encryption"
+
+module Agama
+  module Storage
+    module ConfigConversions
+      module ToJSONConversions
+        # Encryption conversion to JSON hash according to schema.
+        class Encryption < Base
+          # @see Base
+          def self.config_type
+            Configs::Encryption
+          end
+
+          # @see Base#convert
+          # @return [Hash, String, nil]
+          def convert
+            return unless config.method
+
+            super || convert_swap_encryption
+          end
+
+        private
+
+          # @see Base#conversions
+          def conversions
+            method = config.method
+
```

Comment:
Don't we have support for the method `tpm_fde`?

Reply:
Not yet. It is not supported in the schema neither. The *read_only* property for Btrfs is another missing property. If you don't mind, I will add them in a separate PR, because they have nothing to do with the JSON conversion.

Reply:
Ok. Let's approve this then to move on.